src/hardware/pounder/mod.rs

```diff
@@ -91,48 +92,139 @@ impl From<Channel> for GpioPin {
     }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct DdsChannelState {
-    pub phase_offset: f32,
-    pub frequency: f32,
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, MiniconfAtomic)]
+pub struct DdsClockConfig {
+    pub multiplier: u8,
+    pub reference_clock: f32,
+    pub external_clock: bool,
+}
+
+impl Default for DdsClockConfig {
+    fn default() -> Self {
+        Self {
+            multiplier: super::design_parameters::DDS_MULTIPLIER,
+            reference_clock: super::design_parameters::DDS_REF_CLK.to_Hz()
+                as f32,
+            external_clock: false,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct PDHLockGeneratorConfig {
+    pub clock_config: DdsClockConfig,
+    pub ch: [PDHChannel; 2],
+}
+
+impl Default for PDHLockGeneratorConfig {
+    fn default() -> Self {
+        Self {
+            clock_config: DdsClockConfig::default(),
+            ch: [PDHChannel::default(); 2],
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct PDHChannel {
```

Comment:
It is based on the parameters that can be controlled independently in a PDH servo loop.
![image](https://user-images.githubusercontent.com/54844539/187879688-a6db17a1-19f8-401c-b999-0e83f00d1bb6.png)
For example, since the oscillator is tied to both the phase modulator and the mixer (through phase shifter), I tied both DDS of the same InOut pair the same.

Reply:
I know how PDH works. But it's a bad idea to assume that this is the only application of Pounder.
Also since you don't control the relative timing of the frequency updates between in and out, the relative phase isn't deterministic and your assumptions break.